diff -ur a/flint.h b/flint.h
--- a/flint.h	2014-09-25 00:26:52.000000000 +0400
+++ b/flint.h	2014-10-04 09:28:27.000000000 +0400
@@ -291,7 +291,7 @@
 
 #define TMP_ALLOC(size) \
    ((size) > 8192 ? \
-      (__tpx = alloca(sizeof(__tmp_t)), \
+      (__tpx = (__tmp_t *) alloca(sizeof(__tmp_t)), \
        __tpx->next = __tmp_root, \
        __tmp_root = __tpx, \
        __tpx->block = flint_malloc(size)) : \
@@ -367,4 +367,3 @@
 #endif
 
 #endif
-
